asgiref/typing.py

```diff
@@ -189,6 +189,7 @@ class WebSocketResponseBodyEvent(TypedDict):
 class WebSocketDisconnectEvent(TypedDict):
     type: Literal["websocket.disconnect"]
     code: int
+    reason: Optional[str]
```

Comment:
@andrewgodwin Can we start using `NotRequired` on those?

Reply:
Unfortunately that only appears in 3.11...